src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -30,47 +33,40 @@
     @Autowired
     private MongoDatabase database;
 
-    @Override
-    public List<User> findAll(Integer pageNumber, Integer pageCount) throws DataAccessException {
-        if (pageNumber == null){
-            pageNumber = PAGE_NUMBER;
-        }
-        if (pageCount == null){
-            pageCount = PAGE_COUNT;
-        }
-        List<User> list = new ArrayList();
+    private User mapUser(Document userDoc){
+        User user = new User();
+        DateFormat format = new SimpleDateFormat("EEE MMM d HH:mm:ss z yyyy");
+        Date createDate = null;
+        Date modifiedDate = null;
         try {
-            MongoCollection<Document> collection = database.getCollection("users");
-            MongoCursor<Document> iterator = collection.find().skip(pageCount * (pageNumber - 1)).limit(pageCount).iterator();
-            while (iterator.hasNext()) {
-                String jsonInString = iterator.next().toJson();
-                Gson g = new GsonBuilder().create();
-                User user = g.fromJson(jsonInString, User.class);
-                list.add(user);
+            if (userDoc.get("created_date") != null){
+            createDate = format.parse(userDoc.get("created_date").toString());
```

Comment:
User doc should not have a created_date field. It should be taken from the doc's object id